src/components/user/UserProfile.js

```diff
@@ -9,13 +9,48 @@ import {
   Button,
 } from '@mui/material';
 
-import { Avatar } from '@mui/material';
+import { Avatar, Alert } from '@mui/material';
 import { Link } from 'react-router-dom';
 import { useAuth } from '@akosasante/react-auth-context';
 
 export default function PersonalProfile() {
+  const [purchasedAlbums, setPurchasedAlbums] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
   const { user } = useAuth();
 
+  const fetchPurchasedAlbums = useCallback(async () => {
+    try {
+      const response = await axiosInstance.get(`/users/showMe/withAlbums`);
+      console.log('this is what we get', response.data.user.purchasedAlbums);
+      setPurchasedAlbums(response.data.user.purchasedAlbums);
+      setLoading(false);
+    } catch {
+      console.error('Error fetching purchased albums:', error);
+      setError('Error fetching albums for this user. Please try again later.');
+      setLoading(false);
+    }
+  }, [user.userId]);
```

Comment:
Since we use the `error` state variable, this callback hook needs to know when it changes, so our linter is warning us that it's missing from the dependency array. 

At the same time, we don't use `user.userId` inside the useCallback hook so our linter will tell us that this does not need to be in the dependency array at all actually.:

```suggestion
  }, [error]);
```